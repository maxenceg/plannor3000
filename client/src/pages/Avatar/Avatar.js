// @flow
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import TextField from '@material-ui/core/TextField';

import { Content, Container } from './Avatar.style';

// type Props = {
//   fetchUser: (username: string) => void,
//   intl: IntlShape,
//   push: (pathName: string) => Dispatch<*>,
//   updateUsername: (value: string) => void,
//   userAvatarUrl: string,
//   username: string,
// };

class Avatar extends React.Component {
  onInputChange = event => {
    const { updateUsername } = this.props;
    updateUsername(event.target.value);
  };

  fetchUser = () => {
    const { fetchUser, username } = this.props;
    fetchUser(username);
  };

  navigateTo = path => () => {
    const { push } = this.props;
    push(path);
  };

  render() {
    const { intl, userAvatarUrl } = this.props;
    const { formatMessage } = intl;

    return (
      <Container>
        <Content>
          <Button tabIndex={0} onClick={this.navigateTo('/')} variant="contained" color="primary">
            <FormattedMessage id="page.back" />
          </Button>
          <Typography variant="body1">
            <FormattedMessage id="page.api-to-translate-example" />
          </Typography>
          <br />
          <TextField
            fullWidth
            id="github-avatar-input"
            label={formatMessage({ id: 'page.add-github-username' })}
            type="text"
            margin="normal"
            onChange={this.onInputChange}
          />
          <br />
          <Button onClick={this.fetchUser} variant="contained" color="primary">
            <FormattedMessage id="page.fetch-github-avatar" />
          </Button>
          {userAvatarUrl && <img src={userAvatarUrl} alt="user avatar" />}
        </Content>
      </Container>
    );
  }
}

export default Avatar;
