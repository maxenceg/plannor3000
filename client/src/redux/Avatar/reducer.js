// @flow
import { UPDATE_USERNAME, USER_FETCH_SUCCESS } from './constant';

const initialState = {
  userAvatarUrl: null,
  username: null,
};

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case UPDATE_USERNAME:
      return {
        ...state,
        username: action.payload.username,
      };
    case USER_FETCH_SUCCESS:
      return {
        ...state,
        userAvatarUrl: action.payload.user.avatar_url,
      };
    default:
      return state;
  }
}
