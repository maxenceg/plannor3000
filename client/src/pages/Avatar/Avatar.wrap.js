// @flow
import { push } from 'connected-react-router';
import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';

import { actions } from 'redux/Avatar';
import Avatar from './Avatar';

const mapStateToProps = state => ({
  username: state.avatar.username,
  userAvatarUrl: state.avatar.userAvatarUrl,
});

const mapDispatchToProps = dispatch => ({
  fetchUser: username => dispatch(actions.fetchUserRequest(username)),
  push: pathName => dispatch(push(pathName)),
  updateUsername: username => dispatch(actions.updateUsername(username)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(injectIntl(Avatar));
