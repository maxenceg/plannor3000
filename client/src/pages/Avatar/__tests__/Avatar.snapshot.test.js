// @flow
import React from 'react';

import intlShape from '__mocks__/intlShape';
import createComponentWithIntl from 'services/i18n/create-component-with-intl';
import Avatar from '../Avatar';

describe('The page component', () => {
  it('should render correctly', () => {
    const props = {
      intl: intlShape,
      fetchUser: jest.fn(),
      push: jest.fn(),
      updateUsername: jest.fn(),
      userAvatarUrl: 'https://avatars1.githubusercontent.com/u/9215726?v=4',
      username: 'juste_leblanc',
    };
    const tree = createComponentWithIntl(<Avatar {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
