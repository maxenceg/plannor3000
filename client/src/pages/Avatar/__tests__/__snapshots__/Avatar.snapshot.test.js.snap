// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The page component should render correctly 1`] = `
<div
  className="Avatarstyle__Container-vc2bed-0 fWkIzP"
>
  <div
    className="Avatarstyle__Content-vc2bed-1 iSRiGY"
  >
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        <span>
          Retour
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-99"
      />
    </button>
    <p
      className="MuiTypography-root-30 MuiTypography-body1-39"
    >
      <span>
        Ce message est traduit en utilisant directement l'API react-intl
      </span>
    </p>
    <br />
    <div
      className="MuiFormControl-root-65 MuiFormControl-marginNormal-66 MuiFormControl-fullWidth-68"
    >
      <div
        className="MuiInputBase-root-82 MuiInput-root-69 MuiInput-underline-73 MuiInputBase-fullWidth-91 MuiInput-fullWidth-76 MuiInputBase-formControl-83 MuiInput-formControl-70"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input-92 MuiInput-input-77"
          disabled={false}
          id="github-avatar-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
      </div>
    </div>
    <br />
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        <span>
          Affiche l'avatar github
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-99"
      />
    </button>
    <img
      alt="user avatar"
      src="https://avatars1.githubusercontent.com/u/9215726?v=4"
    />
  </div>
</div>
`;
